omnibus/files/private-chef-ctl-commands/server_admins.rb

```diff
@@ -0,0 +1,170 @@
+    STDERR.puts msg
+    raise SystemExit.new(1, msg)
+  end
+
+  username = cmd_args[0]
+
+  if username.downcase == 'pivotal'
+    msg = "You cannot remove the base superuser pivotal from server-admins."
+    STDERR.puts msg
+    raise SystemExit.new(1, msg)
+  end
+
+  bifrost_config = running_service_config('oc_bifrost')
+  superuser_id = bifrost_config['superuser_id']
+  vip = bifrost_config['vip']
+  port = bifrost_config['port']
+
+  db = setup_erchef_db
+  server_admins_authz_id = get_server_admins_authz_id(db)
+  user_authz_id = get_user_authz_id(db, username)
+
+  # put the user in the server-admins authz group
+  headers = {
+    :content_type => :json,
+    :accept => :json,
+    'X-Ops-Requesting-Actor-Id' => superuser_id
+  }
+
+  base_url = "http://#{vip}:#{port}"
+
+  RestClient.delete("#{base_url}/groups/#{server_admins_authz_id}/actors/#{user_authz_id}", headers)
+
+  puts "User #{username} was removed from server-admins. This user can no longer list, read, and create users for this Chef Server except for where they have default permissions (such as within an org)."
+
+end
+
+add_command_under_category "list-server-admins", "server-admins", "List users that have server-admins permissions.", 2 do
+  cmd_args = ARGV[3..-1]
+  if cmd_args.length != 0
+    msg = "This command does not accept arguments."
+    STDERR.puts msg
+    raise SystemExit.new(1, msg)
+  end
+
+  bifrost_config = running_service_config('oc_bifrost')
+  superuser_id = bifrost_config['superuser_id']
+  vip = bifrost_config['vip']
+  port = bifrost_config['port']  
+
+  db = setup_erchef_db
+  server_admins_authz_id = get_server_admins_authz_id(db)
+  
+  # get all the user authz_ids for all memebers of the server-admins authz group
+  headers = {
+    :content_type => :json,
+    :accept => :json,
+    'X-Ops-Requesting-Actor-Id' => superuser_id
+  }
+
+  base_url = "http://#{vip}:#{port}"
+
+  results = JSON.parse(RestClient.get("#{base_url}/groups/#{server_admins_authz_id}", headers))
+
+  # get the user's authz id
+  users = db.exec_params("SELECT * from USERS WHERE authz_id IN #{create_sql_collection_string(results['actors'])}")
+  users.each do |user|
+    puts user['username']
+  end
+end
+
+# returns string in format '('item1','item2',...,'itemN')'
+def create_sql_collection_string(arr)
+  str = arr.join("','")
+  "('#{str}')"
+end
+
+def setup_erchef_db
+  erchef_config = running_service_config('opscode-erchef')
+  pg_config = running_service_config('postgresql')
+  ::PGconn.open('user' => erchef_config['sql_user'],
+                'host' => pg_config['vip'],
+                'password' => erchef_config['sql_password'],
+                'port' => pg_config['port'],
+                'dbname' => 'opscode_chef')
+end
+
+def get_server_admins_authz_id(db)
+  server_admins_erchef_group = db.exec_params("SELECT authz_id FROM groups WHERE name='server-admins' AND org_id='#{PLACEHOLDER_GLOBAL_ORG_ID}'")
+
+  if server_admins_erchef_group.ntuples != 1
```

Comment:
Add check for zero  and tell them to run chef-server-ctl upgrade before proceeding.